diff -Nru mico/MakeVars.in mico.new/MakeVars.in
--- mico/MakeVars.in	Mon Nov 15 23:17:11 1999
+++ mico.new/MakeVars.in	Sun Nov 14 21:10:34 1999
@@ -22,11 +22,11 @@
 #
 
 prefix		= @prefix@
+mandir		= @mandir@
 exec_prefix	= @exec_prefix@
 SRCDIR		= @ABSSRCDIR@
 INSTDIR		= $(exec_prefix)
 SHARED_INSTDIR	= $(prefix)
-
 
 HAVE_REPO	= @HAVE_REPO@
 HAVE_JDK	= @HAVE_JDK@
diff -Nru mico/Makefile mico.new/Makefile
--- mico/Makefile	Sun Jun 27 06:30:29 1999
+++ mico.new/Makefile	Mon Nov 15 23:11:13 1999
@@ -59,16 +59,16 @@
 install:
 	for i in $(INSTALLDIRS); do $(MAKE) -C $$i install || exit 1; done
 	if test -f doc/doc.ps; then \
-		$(IDIRCMD) $(SHARED_INSTDIR)/doc/mico; \
-		$(IMANCMD) doc/doc.ps $(SHARED_INSTDIR)/doc/mico/manual.ps; \
+		$(IDIRCMD) $(DESTDIR)$(SHARED_INSTDIR)/doc/mico; \
+		$(IMANCMD) doc/doc.ps $(DESTDIR)$(SHARED_INSTDIR)/doc/mico/manual.ps; \
 	fi
 	-ldconfig
 
 install-cd:
 	for i in $(INSTALLDIRS); do $(MAKE) -C $$i install-cd || exit 1; done
 	if test -f doc/doc.ps; then \
-		$(IDIRCMD) $(SHARED_INSTDIR)/doc/mico; \
-		$(IMANCMD) doc/doc.ps $(SHARED_INSTDIR)/doc/mico/manual.ps; \
+		$(IDIRCMD) $(DESTDIR)$(SHARED_INSTDIR)/doc/mico; \
+		$(IMANCMD) doc/doc.ps $(DESTDIR)$(SHARED_INSTDIR)/doc/mico/manual.ps; \
 	fi
 
 dist:
diff -Nru mico/admin/Makefile mico.new/admin/Makefile
--- mico/admin/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/admin/Makefile	Sun Nov 14 21:10:34 1999
@@ -37,21 +37,22 @@
 	chmod +x $@
 
 install:
-	$(IDIRCMD) $(INSTDIR)/lib
-	$(ISCRIPTCMD) mico-setup.sh $(INSTDIR)/lib
-	$(ISCRIPTCMD) mico-setup.csh $(INSTDIR)/lib
-	$(IDIRCMD) $(INSTDIR)/bin
-	test -f mico-c++   && $(ISCRIPTCMD) mico-c++   $(INSTDIR)/bin
-	test -f mico-ld    && $(ISCRIPTCMD) mico-ld    $(INSTDIR)/bin
-	test -f mico-ar    && $(ISCRIPTCMD) mico-ar    $(INSTDIR)/bin
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/lib
+	$(ISCRIPTCMD) mico-setup.sh $(DESTDIR)$(INSTDIR)/lib
+	$(ISCRIPTCMD) mico-setup.csh $(DESTDIR)$(INSTDIR)/lib
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/bin
+	test -f mico-c++   && $(ISCRIPTCMD) mico-c++   $(DESTDIR)$(INSTDIR)/bin
+	test -f mico-ld    && $(ISCRIPTCMD) mico-ld    $(DESTDIR)$(INSTDIR)/bin
+	test -f mico-ar    && $(ISCRIPTCMD) mico-ar    $(DESTDIR)$(INSTDIR)/bin
 ifeq ($(HAVE_DYNAMIC), yes)
-	test -f mico-shc++ && $(ISCRIPTCMD) mico-shc++ $(INSTDIR)/bin
-	test -f mico-shld  && $(ISCRIPTCMD) mico-shld  $(INSTDIR)/bin 
+	test -f mico-shc++ && $(ISCRIPTCMD) mico-shc++ $(DESTDIR)$(INSTDIR)/bin
+	test -f mico-shld  && $(ISCRIPTCMD) mico-shld  $(DESTDIR)$(INSTDIR)/bin 
 endif
 
 install-cd: install
-	$(ISCRIPTCMD) mico-setup.sh.in $(INSTDIR)/lib
-	$(ISCRIPTCMD) mico-setup.csh.in $(INSTDIR)/lib
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/lib
+	$(ISCRIPTCMD) mico-setup.sh.in $(DESTDIR)$(INSTDIR)/lib
+	$(ISCRIPTCMD) mico-setup.csh.in $(DESTDIR)$(INSTDIR)/lib
 
 clean:
 #	rm -f mico-setup.sh mico-setup.csh
diff -Nru mico/auxdir/Makefile mico.new/auxdir/Makefile
--- mico/auxdir/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/auxdir/Makefile	Sun Nov 14 21:10:34 1999
@@ -200,12 +200,12 @@
 	-$(RM) *.orig
 
 install:
-	$(IDIRCMD) $(INSTDIR)/lib
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/lib
 ifeq ($(HAVE_SHARED), yes)
-	@for i in $(SHARED_LIBS); do $(ILIBCMD) $$i $(INSTDIR)/lib; done
+	@for i in $(SHARED_LIBS); do $(ILIBCMD) $$i $(DESTDIR)$(INSTDIR)/lib; done
 endif
 ifeq ($(HAVE_STATIC), yes)
-	@for i in $(STATIC_LIBS); do $(ILIBCMD) $$i $(INSTDIR)/lib; done
+	@for i in $(STATIC_LIBS); do $(ILIBCMD) $$i $(DESTDIR)$(INSTDIR)/lib; done
 endif
 
 install-cd: install
diff -Nru mico/coss/Makefile mico.new/coss/Makefile
--- mico/coss/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/coss/Makefile	Sun Nov 14 21:10:34 1999
@@ -157,15 +157,15 @@
 	$(RM) *.a *.$(SOEXT)
 
 install:
-	$(IDIRCMD) $(INSTDIR)/lib
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/lib
 ifneq ($(strip $(SUBDIRS)),)
 ifeq ($(HAVE_SHARED_EXCEPTS), yes)
-	$(ILIBCMD) libmicocoss$(VERSION).$(SOEXT) $(INSTDIR)/lib
+	$(ILIBCMD) libmicocoss$(VERSION).$(SOEXT) $(DESTDIR)$(INSTDIR)/lib
 ifeq ($(HAVE_STATIC), yes)
-	$(ILIBCMD) libmicocoss$(VERSION).a $(INSTDIR)/lib
+	$(ILIBCMD) libmicocoss$(VERSION).a $(DESTDIR)$(INSTDIR)/lib
 endif
 else
-	$(ILIBCMD) libmicocoss$(VERSION).a $(INSTDIR)/lib
+	$(ILIBCMD) libmicocoss$(VERSION).a $(DESTDIR)$(INSTDIR)/lib
 endif
 	for i in $(SUBDIRS); do $(MAKE) -C $$i install || exit 1; done	
 endif
diff -Nru mico/coss/events/Makefile mico.new/coss/events/Makefile
--- mico/coss/events/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/coss/events/Makefile	Sun Nov 14 21:10:34 1999
@@ -77,8 +77,8 @@
 	-$(RM) ../../include/mico/CosEventComm.h
 
 install:
-	$(IDIRCMD) $(INSTDIR)/bin
-	$(IBINCMD) eventd $(INSTDIR)/bin
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/bin
+	$(IBINCMD) eventd $(DESTDIR)$(INSTDIR)/bin
 
 ifeq (.depend, $(wildcard .depend))
 include .depend
diff -Nru mico/coss/naming/Makefile mico.new/coss/naming/Makefile
--- mico/coss/naming/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/coss/naming/Makefile	Sun Nov 14 21:10:34 1999
@@ -71,9 +71,9 @@
 		CosNaming_skel.cc
 
 install:
-	$(IDIRCMD) $(INSTDIR)/bin
-	$(IBINCMD) nsd $(INSTDIR)/bin
-	$(IBINCMD) nsadmin $(INSTDIR)/bin
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/bin
+	$(IBINCMD) nsd $(DESTDIR)$(INSTDIR)/bin
+	$(IBINCMD) nsadmin $(DESTDIR)$(INSTDIR)/bin
 
 ifeq (.depend, $(wildcard .depend))
 include .depend
diff -Nru mico/coss/property/Makefile mico.new/coss/property/Makefile
--- mico/coss/property/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/coss/property/Makefile	Sun Nov 14 21:10:34 1999
@@ -68,8 +68,8 @@
 	-$(RM) propertyd
 
 install:
-	$(IDIRCMD) $(INSTDIR)/bin
-	$(IBINCMD) propertyd $(INSTDIR)/bin
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/bin
+	$(IBINCMD) propertyd $(DESTDIR)$(INSTDIR)/bin
 
 ifeq (.depend, $(wildcard .depend))
 include .depend
diff -Nru mico/coss/relship/Makefile mico.new/coss/relship/Makefile
--- mico/coss/relship/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/coss/relship/Makefile	Mon Nov 15 00:59:07 1999
@@ -164,18 +164,18 @@
 	-$(RM) ../../include/mico/random.h
 
 install:
-	$(IDIRCMD) $(INSTDIR)/bin
-	$(IBINCMD) traversald $(INSTDIR)/bin
-	$(IDIRCMD) $(INSTDIR)/bin
-	$(IBINCMD) noded $(INSTDIR)/bin
-	$(IDIRCMD) $(INSTDIR)/bin
-	$(IBINCMD) containmentd $(INSTDIR)/bin
-	$(IDIRCMD) $(INSTDIR)/bin
-	$(IBINCMD) referenced $(INSTDIR)/bin
-	$(IDIRCMD) $(INSTDIR)/bin
-	$(IBINCMD) randomd $(INSTDIR)/bin
-	$(IDIRCMD) $(INSTDIR)/bin
-	$(IBINCMD) proxyd $(INSTDIR)/bin
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/bin
+	$(IBINCMD) traversald $(DESTDIR)$(INSTDIR)/bin
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/bin
+	$(IBINCMD) noded $(DESTDIR)$(INSTDIR)/bin
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/bin
+	$(IBINCMD) containmentd $(DESTDIR)$(INSTDIR)/bin
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/bin
+	$(IBINCMD) referenced $(DESTDIR)$(INSTDIR)/bin
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/bin
+	$(IBINCMD) randomd $(DESTDIR)$(INSTDIR)/bin
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/bin
+	$(IBINCMD) proxyd $(DESTDIR)$(INSTDIR)/bin
 
 
 ifeq (.depend, $(wildcard .depend))
diff -Nru mico/coss/time/Makefile mico.new/coss/time/Makefile
--- mico/coss/time/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/coss/time/Makefile	Mon Nov 15 00:59:29 1999
@@ -67,8 +67,8 @@
 	-$(RM) ../../include/mico/CosTime.h
 
 install:
-	$(IDIRCMD) $(INSTDIR)/bin
-	$(IBINCMD) timed $(INSTDIR)/bin
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/bin
+	$(IBINCMD) timed $(DESTDIR)$(INSTDIR)/bin
 
 ifeq (.depend, $(wildcard .depend))
 include .depend
diff -Nru mico/coss/trader/Makefile mico.new/coss/trader/Makefile
--- mico/coss/trader/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/coss/trader/Makefile	Sun Nov 14 21:10:34 1999
@@ -85,8 +85,8 @@
 	-$(RM) $(GENERATED) CosTrading_skel.cc CosTradingRepos_skel.cc
 
 install:
-	$(IDIRCMD) $(INSTDIR)/bin
-	$(IBINCMD) traderd $(INSTDIR)/bin
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/bin
+	$(IBINCMD) traderd $(DESTDIR)$(INSTDIR)/bin
 
 generate:
 	$(BISON) -d -t -v -o yacc.c yacc.yy
diff -Nru mico/cpp/Makefile mico.new/cpp/Makefile
--- mico/cpp/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/cpp/Makefile	Sun Nov 14 21:10:34 1999
@@ -56,8 +56,8 @@
 
 
 install: all
-	$(IDIRCMD) $(INSTDIR)/bin
-	$(IBINCMD) mico-cpp $(INSTDIR)/bin
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/bin
+	$(IBINCMD) mico-cpp $(DESTDIR)$(INSTDIR)/bin
 
 install-cd: install
 
diff -Nru mico/daemon/Makefile mico.new/daemon/Makefile
--- mico/daemon/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/daemon/Makefile	Sun Nov 14 21:10:34 1999
@@ -48,8 +48,8 @@
 
 
 install: all
-	$(IDIRCMD) $(INSTDIR)/bin
-	$(IBINCMD) micod $(INSTDIR)/bin
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/bin
+	$(IBINCMD) micod $(DESTDIR)$(INSTDIR)/bin
 
 install-cd: install
 
diff -Nru mico/demo/MakeVars mico.new/demo/MakeVars
--- mico/demo/MakeVars	Mon Nov 15 23:17:11 1999
+++ mico.new/demo/MakeVars	Sun Nov 14 21:10:34 1999
@@ -78,14 +78,14 @@
 endif
 
 install:
-	$(IDIRCMD) $(INSTDIR)/doc/mico/examples/$(INSTALL_DIR)
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/doc/mico/examples/$(INSTALL_DIR)
 	test -z "$(INSTALL_SRCS)" || \
 	  eval 'for i in $(INSTALL_SRCS); do \
-	    $(ISRCCMD) $$i $(INSTDIR)/doc/mico/examples/$(INSTALL_DIR); \
+	    $(ISRCCMD) $$i $(DESTDIR)$(INSTDIR)/doc/mico/examples/$(INSTALL_DIR); \
 	  done'
 	test -z "$(INSTALL_SCRIPTS)" || \
 	  eval 'for i in $(INSTALL_SCRIPTS); do \
-	    $(ISCRIPTCMD) $$i $(INSTDIR)/doc/mico/examples/$(INSTALL_DIR); \
+	    $(ISCRIPTCMD) $$i $(DESTDIR)$(INSTDIR)/doc/mico/examples/$(INSTALL_DIR); \
 	  done'
 
 ifeq (.depend, $(wildcard .depend))
diff -Nru mico/demo/Makefile mico.new/demo/Makefile
--- mico/demo/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/demo/Makefile	Sun Nov 14 21:10:34 1999
@@ -44,12 +44,12 @@
 	for i in $(DIRS); do $(MAKE) -C $$i clean || exit 1; done
 
 install:
-	$(IDIRCMD) $(INSTDIR)/doc/mico/examples
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/doc/mico/examples
 	sed -e s/@MICO_VERSION@/$(VERSION)/g \
 		< MakeVars.install > MakeVars.install.ver
-	$(ISRCCMD) MakeVars.install.ver $(INSTDIR)/doc/mico/examples/MakeVars
+	$(ISRCCMD) MakeVars.install.ver $(DESTDIR)$(INSTDIR)/doc/mico/examples/MakeVars
 	rm -f MakeVars.install.ver
-	$(ISRCCMD) README $(INSTDIR)/doc/mico/examples
+	$(ISRCCMD) README $(DESTDIR)$(INSTDIR)/doc/mico/examples
 	for i in $(INST_DIRS); do $(MAKE) -C $$i install || exit 1; done
 
 install-cd: install
diff -Nru mico/demo/boa/Makefile mico.new/demo/boa/Makefile
--- mico/demo/boa/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/demo/boa/Makefile	Mon Nov 15 05:02:58 1999
@@ -31,7 +31,7 @@
 	for i in $(DIRS); do $(MAKE) -C $$i clean || exit 1; done
 
 install:
-	$(IDIRCMD) $(INSTDIR)/doc/mico/examples/boa
-	$(ISRCCMD) Makefile $(INSTDIR)/doc/mico/examples/boa
-	$(ISRCCMD) README $(INSTDIR)/doc/mico/examples/boa
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/doc/mico/examples/boa
+	$(ISRCCMD) Makefile $(DESTDIR)$(INSTDIR)/doc/mico/examples/boa
+	$(ISRCCMD) README $(DESTDIR)$(INSTDIR)/doc/mico/examples/boa
 	for i in $(DIRS); do $(MAKE) -C $$i install || exit 1; done
diff -Nru mico/demo/dispatcher/Makefile mico.new/demo/dispatcher/Makefile
--- mico/demo/dispatcher/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/demo/dispatcher/Makefile	Mon Nov 15 05:03:46 1999
@@ -27,8 +27,8 @@
 	rm -f *.o core *~ .depend
 
 install:
-	$(IDIRCMD) $(INSTDIR)/doc/mico/examples/dispatcher
-	$(ISRCCMD) Makefile $(INSTDIR)/doc/mico/examples/dispatcher
-	$(ISRCCMD) README $(INSTDIR)/doc/mico/examples/dispatcher
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/doc/mico/examples/dispatcher
+	$(ISRCCMD) Makefile $(DESTDIR)$(INSTDIR)/doc/mico/examples/dispatcher
+	$(ISRCCMD) README $(DESTDIR)$(INSTDIR)/doc/mico/examples/dispatcher
 	test -z "$(DIRS)" || \
 	  eval 'for i in $(DIRS); do $(MAKE) -C $$i install || exit 1; done'
diff -Nru mico/demo/interop/Makefile mico.new/demo/interop/Makefile
--- mico/demo/interop/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/demo/interop/Makefile	Mon Nov 15 05:04:23 1999
@@ -30,7 +30,7 @@
 	for i in $(DIRS); do $(MAKE) -C $$i clean || exit 1; done
 
 install:
-	$(IDIRCMD) $(INSTDIR)/doc/mico/examples/interop
-	$(ISRCCMD) Makefile $(INSTDIR)/doc/mico/examples/interop
-	$(ISRCCMD) README $(INSTDIR)/doc/mico/examples/interop
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/doc/mico/examples/interop
+	$(ISRCCMD) Makefile $(DESTDIR)$(INSTDIR)/doc/mico/examples/interop
+	$(ISRCCMD) README $(DESTDIR)$(INSTDIR)/doc/mico/examples/interop
 	for i in $(DIRS); do $(MAKE) -C $$i install || exit 1; done
diff -Nru mico/demo/obv/Makefile mico.new/demo/obv/Makefile
--- mico/demo/obv/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/demo/obv/Makefile	Mon Nov 15 05:04:46 1999
@@ -13,7 +13,7 @@
 	rm -f *.o core *~ .depend
 
 install:
-	$(IDIRCMD) $(INSTDIR)/doc/mico/examples/obv
-	$(ISRCCMD) Makefile $(INSTDIR)/doc/mico/examples/obv
-	$(ISRCCMD) README $(INSTDIR)/doc/mico/examples/obv
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/doc/mico/examples/obv
+	$(ISRCCMD) Makefile $(DESTDIR)$(INSTDIR)/doc/mico/examples/obv
+	$(ISRCCMD) README $(DESTDIR)$(INSTDIR)/doc/mico/examples/obv
 	for i in $(DIRS); do $(MAKE) -C $$i install || exit 1; done
diff -Nru mico/demo/poa/Makefile mico.new/demo/poa/Makefile
--- mico/demo/poa/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/demo/poa/Makefile	Mon Nov 15 05:05:01 1999
@@ -13,7 +13,7 @@
 	rm -f *.o core *~ .depend
 
 install:
-	$(IDIRCMD) $(INSTDIR)/doc/mico/examples/poa
-	$(ISRCCMD) Makefile $(INSTDIR)/doc/mico/examples/poa
-	$(ISRCCMD) README $(INSTDIR)/doc/mico/examples/poa
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/doc/mico/examples/poa
+	$(ISRCCMD) Makefile $(DESTDIR)$(INSTDIR)/doc/mico/examples/poa
+	$(ISRCCMD) README $(DESTDIR)$(INSTDIR)/doc/mico/examples/poa
 	for i in $(DIRS); do $(MAKE) -C $$i install || exit 1; done
diff -Nru mico/demo/redlich/Makefile mico.new/demo/redlich/Makefile
--- mico/demo/redlich/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/demo/redlich/Makefile	Mon Nov 15 05:05:20 1999
@@ -33,7 +33,7 @@
 	for i in $(DIRS); do $(MAKE) -C $$i clean || exit 1; done
 
 install:
-	$(IDIRCMD) $(INSTDIR)/doc/mico/examples/redlich
-	$(ISRCCMD) Makefile $(INSTDIR)/doc/mico/examples/redlich
-	$(ISRCCMD) README $(INSTDIR)/doc/mico/examples/redlich
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/doc/mico/examples/redlich
+	$(ISRCCMD) Makefile $(DESTDIR)$(INSTDIR)/doc/mico/examples/redlich
+	$(ISRCCMD) README $(DESTDIR)$(INSTDIR)/doc/mico/examples/redlich
 	for i in $(DIRS); do $(MAKE) -C $$i install || exit 1; done
diff -Nru mico/demo/services/Makefile mico.new/demo/services/Makefile
--- mico/demo/services/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/demo/services/Makefile	Mon Nov 15 05:05:34 1999
@@ -13,7 +13,7 @@
 	rm -f *.o core *~ .depend
 
 install:
-	$(IDIRCMD) $(INSTDIR)/doc/mico/examples/services
-	$(ISRCCMD) Makefile $(INSTDIR)/doc/mico/examples/services
-	$(ISRCCMD) README $(INSTDIR)/doc/mico/examples/services
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/doc/mico/examples/services
+	$(ISRCCMD) Makefile $(DESTDIR)$(INSTDIR)/doc/mico/examples/services
+	$(ISRCCMD) README $(DESTDIR)$(INSTDIR)/doc/mico/examples/services
 	for i in $(DIRS) stream-sound; do $(MAKE) -C $$i install || exit 1; done
diff -Nru mico/idl/Makefile mico.new/idl/Makefile
--- mico/idl/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/idl/Makefile	Sun Nov 14 21:10:34 1999
@@ -79,8 +79,8 @@
 
 
 install: all
-	$(IDIRCMD) $(INSTDIR)/bin
-	$(IBINCMD) idl $(INSTDIR)/bin
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/bin
+	$(IBINCMD) idl $(DESTDIR)$(INSTDIR)/bin
 
 install-cd: install
 
diff -Nru mico/imr/Makefile mico.new/imr/Makefile
--- mico/imr/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/imr/Makefile	Sun Nov 14 21:10:34 1999
@@ -99,8 +99,8 @@
 	-$(RM) .depend TAGS core nohup.out
 
 install:
-	$(IDIRCMD) $(INSTDIR)/bin
-	$(IBINCMD) imr $(INSTDIR)/bin
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/bin
+	$(IBINCMD) imr $(DESTDIR)$(INSTDIR)/bin
 
 install-cd: install
 
diff -Nru mico/include/Makefile mico.new/include/Makefile
--- mico/include/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/include/Makefile	Sun Nov 14 21:10:34 1999
@@ -37,14 +37,14 @@
 	echo "#define MICO_BIN_VERSION $(BIN_VER)" >> mico/version.h
 
 install:
-	$(IDIRCMD) $(INSTDIR)/include
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/include
 	for i in *.h; do \
-	  $(IINCCMD) $$i $(INSTDIR)/include; \
+	  $(IINCCMD) $$i $(DESTDIR)$(INSTDIR)/include; \
 	done
 	for i in $(DIRS); do \
-	  $(IDIRCMD) $(INSTDIR)/include/$$i; \
+	  $(IDIRCMD) $(DESTDIR)$(INSTDIR)/include/$$i; \
 	  for j in $$i/*; do \
-	    $(IINCCMD) $$j $(INSTDIR)/include/$$i/; \
+	    $(IINCCMD) $$j $(DESTDIR)$(INSTDIR)/include/$$i/; \
 	  done; \
 	done
 
diff -Nru mico/ir/Makefile mico.new/ir/Makefile
--- mico/ir/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/ir/Makefile	Sun Nov 14 21:10:34 1999
@@ -116,8 +116,8 @@
 	-$(RM) .depend TAGS core nohup.out
 
 install:
-	$(IDIRCMD) $(INSTDIR)/bin
-	$(IBINCMD) ird $(INSTDIR)/bin
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/bin
+	$(IBINCMD) ird $(DESTDIR)$(INSTDIR)/bin
 
 install-cd: install
 
diff -Nru mico/man/Makefile mico.new/man/Makefile
--- mico/man/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/man/Makefile	Sun Nov 14 21:10:34 1999
@@ -26,9 +26,9 @@
 
 install:
 	for i in $(DIRS); do \
-	  $(IDIRCMD) $(SHARED_INSTDIR)/man/$$i; \
+	  $(IDIRCMD) $(DESTDIR)$(mandir)/$$i; \
 	  for j in $$i/*.[1-9]; do $(IMANCMD) $$j \
-		$(SHARED_INSTDIR)/man/$$i/; done; \
+		$(DESTDIR)$(mandir)/$$i/; done; \
 	done
 
 install-cd: install
diff -Nru mico/orb/Makefile mico.new/orb/Makefile
--- mico/orb/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/orb/Makefile	Sun Nov 14 21:10:34 1999
@@ -185,12 +185,12 @@
 	$(RM) *.rpo *.a *.o *~ *.$(SOEXT) .depend
 
 install:
-	$(IDIRCMD) $(INSTDIR)/lib
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/lib
 ifeq ($(HAVE_STATIC), yes)
-	$(ILIBCMD) libmico$(VERSION).a $(INSTDIR)/lib
+	$(ILIBCMD) libmico$(VERSION).a $(DESTDIR)$(INSTDIR)/lib
 endif
 ifeq ($(HAVE_SHARED), yes)
-	$(ILIBCMD) libmico$(VERSION).$(SOEXT) $(INSTDIR)/lib
+	$(ILIBCMD) libmico$(VERSION).$(SOEXT) $(DESTDIR)$(INSTDIR)/lib
 endif
 
 install-cd: install
diff -Nru mico/tools/iordump/Makefile mico.new/tools/iordump/Makefile
--- mico/tools/iordump/Makefile	Mon Nov 15 23:17:11 1999
+++ mico.new/tools/iordump/Makefile	Sun Nov 14 21:10:34 1999
@@ -41,8 +41,8 @@
 	$(POSTLD) $@
 
 install:
-	$(IDIRCMD) $(INSTDIR)/bin
-	$(IBINCMD) iordump $(INSTDIR)/bin
+	$(IDIRCMD) $(DESTDIR)$(INSTDIR)/bin
+	$(IBINCMD) iordump $(DESTDIR)$(INSTDIR)/bin
 
 # cleaning
 
