--- mico/orb/transport.cc~	Sat Aug  7 18:52:42 1999
+++ mico/orb/transport.cc	Sat Aug  7 18:58:27 1999
@@ -713,7 +713,7 @@
 
     CORBA::Long retries = 5;
     while (--retries >= 0) {
-	if (write (UDP_MAGIC_CREQ, UDP_MAGIC_SIZE) != UDP_MAGIC_SIZE)
+	if (write ((void *) UDP_MAGIC_CREQ, UDP_MAGIC_SIZE) != UDP_MAGIC_SIZE)
 	    assert (0);
 	if (collect_replies (300 /*ms*/) > 0)
 	    break;
@@ -818,7 +818,7 @@
 		continue;
 	    if (!strncmp((char*)dgram.buffer(),UDP_MAGIC_CREQ,UDP_MAGIC_SIZE)){
 		// peer resent connect request, but reply was lost
-		if (write (UDP_MAGIC_CREP, UDP_MAGIC_SIZE) != UDP_MAGIC_SIZE)
+		if (write ((void *) UDP_MAGIC_CREP, UDP_MAGIC_SIZE) != UDP_MAGIC_SIZE)
 		    assert (0);
 		continue;
 	    }
@@ -1079,7 +1079,7 @@
     if (!t->connect_server (&raddr))
 	assert (0);
 
-    if (t->write (UDP_MAGIC_CREP, UDP_MAGIC_SIZE) != UDP_MAGIC_SIZE)
+    if (t->write ((void *) UDP_MAGIC_CREP, UDP_MAGIC_SIZE) != UDP_MAGIC_SIZE)
 	assert (0);
 
     return t;
